document-generator-prosecution/src/main/java/uk/gov/companieshouse/document/generator/prosecution/ProsecutionDocumentInfoService.java

```diff
@@ -0,0 +1,180 @@
+ * values and other metadata.
+ * </p>
+ * 
+ *
+ * <h3>Matching</h3> The enum uk.gov.companieshouse.document.generator.api.document.DocumentType,
+ * within the document generator api project, is used to determine whether or not to steer a request
+ * to this service, based on whether the URI matches the regex in
+ * uk.gov.companieshouse.document.generator.api.document.DocumentType.PROSECUTION. </br>
+ * Matching URIs must take the form:
+ * <ul>
+ * <li>Ultimatum: <code>/prosecution/ultimatum{urlOfProsecutionCase}</code></br>
+ * e.g.
+ * <code>/prosecution/ultimatum/company/00066516/prosecution-cases/dd908afef2b8529aab3c680239f5d06717113634</code>
+ * </li>
+ * <li>SJP: <code>/prosecution/sjp{urlOfProsecutionCase}</code></br>
+ * e.g.
+ * <code>/prosecution/sjp/company/00066516/prosecution-cases/dd908afef2b8529aab3c680239f5d06717113634</code>
+ * </li>
+ * </ul>
+ * This service will be matched on the base of this URI, will use the base to further determine
+ * exactly which document type to produce, and then will use the rest of the URI to get the data
+ * that it needs.
+ */
+@Service
+public class ProsecutionDocumentInfoService implements DocumentInfoService {
+    public static final String MODULE_NAME_SPACE = "document-generator-prosecution";
+
+    private static final String LOGGING_RESOURCE_KEY = "resource";
+
+    private static final Logger LOG = LoggerFactory.getLogger(MODULE_NAME_SPACE);
+
+    private static final String MATCH_START_OF_SJP_URI = "/prosecution/sjp/";
+
+    private static final String MATCH_START_OF_ULTIMATUM_URI = "/prosecution/ultimatum/";
+
+    private static final int ULTIMATUM_INDEX_TO_TRIM_FROM =
+                    MATCH_START_OF_ULTIMATUM_URI.length() - 1;
+
+    private static final int SJP_INDEX_TO_TRIM_FROM = MATCH_START_OF_SJP_URI.length() - 1;
+
+    private final UltimatumDocumentInfoBuilderProvider docInfoBuilderProvider;
+
+    private final ProsecutionClient prosecutionClient;
+
+    /**
+     * Constructor, which will be autowired.
+     * 
+     * @param docInfoBuilderProvider
+     * @param prosecutionClient
+     */
+    public ProsecutionDocumentInfoService(
+                    UltimatumDocumentInfoBuilderProvider docInfoBuilderProvider,
+                    ProsecutionClient prosecutionClient) {
+        this.docInfoBuilderProvider = docInfoBuilderProvider;
+        this.prosecutionClient = prosecutionClient;
+    }
+
+    /**
+     * Entry point from Document Generator, asking for a DocumentInfoResponse that contains info to
+     * build a document, such info will include the location of any template, any variables that are
+     * needed by the template, etc - see DocumentInfoResponse for more info.
+     */
+    @Override
+    public DocumentInfoResponse getDocumentInfo(DocumentInfoRequest documentInfoRequest)
+                    throws DocumentInfoException {
+        LOG.info("Started getting document");
```

Comment:
Can you extend this log to include the document type please? This will be in the same log file as the other document generations (e.g. accounts).

Reply:
fixed